libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/CarAppLifecycleOwner.kt

```diff
@@ -134,12 +134,12 @@ internal class CarAppLifecycleOwner : LifecycleOwner {
         }
 
         override fun onActivityDestroyed(activity: Activity) {
+            if (!activitiesCreated.remove(activity)) return
```

Comment:
This should go inside the `~isChangingConfigurations` block right? 

Otherwise it will destroy MapboxNavigation when changing configurations

Reply:
Never mind it makes sense. Also there is a test that proves it works 
```
activityLifecycleCallbacks verify orientation switching does not stop the app lifecycle
```

This needs to be moved before to ensure the old destroyed activity is removed

Reply:
Actually there was a small regression. When there are two activities being destroyed, the first one has `isChangingConfigurations` and the second one doesn't, `MapboxNavigation` will be destroyed, but it shouldn't. I don't know if it is possible in real life, but I fixed that anyway, because this case was working as expected before my PR.